src/widgets/gl_area.rs

```diff
@@ -0,0 +1,215 @@
+struct_Widget!(GLArea);
+
+impl GLArea {
+    /// Creates a new GLArea widget.
+    pub fn new() -> Option<GLArea> {
+        let tmp_pointer = unsafe { ffi::gtk_gl_area_new() };
+        check_pointer!(tmp_pointer, GLArea)
+    }
+
+    /// Retrieves the gdk::GLContext used by area.
+    pub fn get_context() -> Option<gdk::GLContext> {
+        unsafe { from_glib_none(ffi::gtk_gl_area_get_context()) }
+    }
+
+    /// Ensures that the gdk::GLContext used by area is associated with the gtk::GLArea.
+    ///
+    /// This function is automatically called before emitting the "render" signal, and doesn't
+    /// normally need to be called by application code.
+    pub fn make_current(&self) {
+        unsafe { ffi::gtk_gl_area_make_current(GTK_GL_AREA(self.pointer)) }
+    }
+
+    /// Marks the currently rendered data (if any) as invalid, and queues a redraw of the widget,
+    /// ensuring that the "render" signal is emitted during the draw.
+    ///
+    /// This is only needed when the GLArea::set_auto_render() has been called with a false value.
+    /// The default behaviour is to emit "render" on each draw.
+    pub fn queue_render(&self) {
+        unsafe { ffi::gtk_gl_area_queue_render(GTK_GL_AREA(self.pointer)) }
+    }
+
+    /// Ensures that the area framebuffer object is made the current draw and read target, and that
+    /// all the required buffers for the area are created and bound to the frambuffer.
+    ///
+    /// This function is automatically called before emitting the "render" signal, and doesn't
+    /// normally need to be called by application code.
+    pub fn attach_buffers(&self) {
+        unsafe { ffi::gtk_gl_area_attach_buffers(GTK_GL_AREA(self.pointer)) }
+    }
+
+    /// Sets an error on the area which will be shown instead of the GL rendering. This is useful in
+    /// the "create-context" signal if GL context creation fails.
+    /*pub fn set_error(&self, error: glib::Error) {
+        unsafe { ffi::gtk_gl_area_set_error(GTK_GL_AREA(self.pointer), error.pointer) }
+    }
+
+    /// Gets the current error set on the area.
+    pub fn get_error(&self) -> Option<error: glib::Error> {
+        unsafe { check_pointer!(ffi::gtk_gl_area_get_error(GTK_GL_AREA(self.pointer)), glib::Error) }
+    }
+    */
+
+    /// If has_alpha is true the buffer allocated by the widget will have an alpha channel component,
+    /// and when rendering to the window the result will be composited over whatever is below the widget.
+    ///
+    /// If has_alpha is false there will be no alpha channel, and the buffer will fully replace
+    /// anything below the widget.
+    pub fn set_has_alpha(&self, has_alpha: bool) {
+        unsafe { ffi::gtk_gl_area_set_has_alpha(GTK_GL_AREA(self.pointer), to_gboolean(has_alpha)) }
+    }
+
+    /// Returns whether the area has an alpha component.
+    pub fn has_alpha(&self) -> bool {
```

Comment:
`get_` should probably be in the method names as well, or the generator will break this